pkgs/tools/filesystems/ceph/default.nix

```diff
@@ -1,621 +0,0 @@
-          outputs = lib.filter (o: o != "doc") old.outputs;
-        });
-
-        # This is the most recent version of `trustme` that's still compatible with `cryptography` 40.
-        # See https://github.com/NixOS/nixpkgs/issues/359723
-        # and https://github.com/python-trio/trustme/commit/586f7759d5c27beb44da60615a71848eb2a5a490
-        trustme = self.callPackage ./old-python-packages/trustme.nix { };
-
-        fastapi = super.fastapi.overridePythonAttrs (old: {
-          # Flaky test:
-          #     ResourceWarning: Unclosed <MemoryObjectSendStream>
-          # Unclear whether it's flaky in general or only in this overridden package set.
-          doCheck = false;
-        });
-
-        # Ceph does not support `kubernetes` >= 19, see:
-        #     https://github.com/NixOS/nixpkgs/pull/281858#issuecomment-1900324090
-        kubernetes = super.kubernetes.overridePythonAttrs (old: rec {
-          version = "18.20.0";
-          src = fetchFromGitHub {
-            owner = "kubernetes-client";
-            repo = "python";
-            rev = "v${version}";
-            sha256 = "1sawp62j7h0yksmg9jlv4ik9b9i1a1w9syywc9mv8x89wibf5ql1";
-            fetchSubmodules = true;
-          };
-        });
-
-      };
-  };
-
-  boost' = boost183.override {
-    enablePython = true;
-    inherit python;
-  };
-
-  # TODO: split this off in build and runtime environment
-  ceph-python-env = python.withPackages (
-    ps: with ps; [
-      ceph-common
-
-      # build time
-      cython_0
-
-      # debian/control
-      bcrypt
-      cherrypy
-      influxdb
-      jinja2
-      kubernetes
-      natsort
-      numpy
-      pecan
-      prettytable
-      pyjwt
-      pyopenssl
-      python-dateutil
-      pyyaml
-      requests
-      routes
-      scikit-learn
-      scipy
-      setuptools
-      sphinx
-      virtualenv
-      werkzeug
-
-      # src/cephadm/zipapp-reqs.txt
-      markupsafe
-
-      # src/pybind/mgr/requirements-required.txt
-      cryptography
-      jsonpatch
-
-      # src/tools/cephfs/shell/setup.py
-      cmd2
-      colorama
-    ]
-  );
-  inherit (ceph-python-env.python) sitePackages;
-
-  version = "19.2.3";
-  src = fetchurl {
-    url = "https://download.ceph.com/tarballs/ceph-${version}.tar.gz";
-    hash = "sha256-zlgp28C81SZbaFJ4yvQk4ZgYz4K/aZqtcISTO8LscSU=";
-  };
-in
-rec {
-  ceph = stdenv.mkDerivation {
-    pname = "ceph";
-    inherit src version;
-
-    patches = [
-      ./boost-1.85.patch
-
-      (fetchpatch2 {
-        name = "ceph-boost-1.86-uuid.patch";
-        url = "https://github.com/ceph/ceph/commit/01306208eac492ee0e67bff143fc32d0551a2a6f.patch?full_index=1";
-        hash = "sha256-OnDrr72inzGXXYxPFQevsRZImSvI0uuqFHqtFU2dPQE=";
-      })
-
-      # See:
-      # * <https://github.com/boostorg/python/issues/394>
-      # * <https://aur.archlinux.org/cgit/aur.git/commit/?h=ceph&id=8c5cc7d8deec002f7596b6d0860859a0a718f12b>
-      # * <https://github.com/ceph/ceph/pull/60999>
```

Comment:
Comments like the above should not be removed.

Spotting such removals is another good point why refactorings to e.g. `by-name` should be split from feature updates.